test/e2e/scheduling/priorities.go

```diff
@@ -303,19 +303,40 @@ var _ = SIGDescribe("SchedulerPriorities [Serial]", func() {
 		// Apply 10 taints to first node
 		nodeName := nodeList.Items[0].Name
 
-		ginkgo.By("Trying to apply 10 (tolerable) taints on the first node.")
+		// First, create a set of tolerable taints (+tolerations) for the first node.
+		// We immediately defer the removal of these taints because addTaintToNode can
+		// panic and RemoveTaintsOffNode does not return an error if the taint does not exist.
+		// Generate 10 tolerable taints for the first node (and matching tolerations)
+		tolerableTaints := make([]v1.Taint, 0)
 		var tolerations []v1.Toleration
 		for i := 0; i < 10; i++ {
-			testTaint := addRandomTaintToNode(cs, nodeName)
+			testTaint := getRandomTaint()
+			tolerableTaints = append(tolerableTaints, testTaint)
 			tolerations = append(tolerations, v1.Toleration{Key: testTaint.Key, Value: testTaint.Value, Effect: testTaint.Effect})
-			defer e2enode.RemoveTaintOffNode(cs, nodeName, *testTaint)
 		}
+		defer e2enode.RemoveTaintsOffNode(cs, nodeName, tolerableTaints)
+		// Generate 10 intolerable taints for each of the remaining nodes
+		intolerableTaints := make(map[string][]v1.Taint)
+		for i := 1; i < len(nodeList.Items); i++ {
+			nodeTaints := make([]v1.Taint, 0)
+			for i := 0; i < 10; i++ {
+				nodeTaints = append(nodeTaints, getRandomTaint())
+			}
+			intolerableTaints[nodeList.Items[i].Name] = nodeTaints
+			defer e2enode.RemoveTaintsOffNode(cs, nodeList.Items[i].Name, nodeTaints)
+		}
+
+		// Apply the tolerable taints generated above to the first node
+		ginkgo.By("Trying to apply 10 (tolerable) taints on the first node.")
+		for _, taint := range tolerableTaints {
+			addTaintToNode(cs, nodeName, taint)
+		}
+		// Apply the intolerable taints to each of the following nodes
 		ginkgo.By("Adding 10 intolerable taints to all other nodes")
 		for i := 1; i < len(nodeList.Items); i++ {
 			node := nodeList.Items[i]
-			for i := 0; i < 10; i++ {
-				testTaint := addRandomTaintToNode(cs, node.Name)
-				defer e2enode.RemoveTaintOffNode(cs, node.Name, *testTaint)
+			for _, taints := range intolerableTaints[node.Name] {
+				addTaintToNode(cs, node.Name, taints)
```

Comment:
e.g.
```suggestion
			taints := intolerableTaints[node.Name]
			defer e2enode.RemoveTaintsOffNode(cs, node.Name, taints)
			for _, taint := range taints {
				addTaintToNode(cs, node.Name, taint)
```

Reply:
if not the intermediate variable approach I suggested above, I would at least like you to drop the plural from the variable you're using in the for loop; `taints` keeps making me think you're adding a list, but `addTaintToNode` only takes a singular taint, so it looks like there's a bug, but the compiler's happy, and.... anyway please save future maintainers the confusion
```suggestion
			for _, taint := range intolerableTaints[node.Name] {
				addTaintToNode(cs, node.Name, taint)
```

Reply:
I see the confusion, updated to just remove the pluralization